heron/scheduler-core/src/java/com/twitter/heron/scheduler/utils/SchedulerJsonAPIUtils.java

```diff
@@ -0,0 +1,146 @@
+// Copyright 2017 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.scheduler.utils;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.Arrays;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.twitter.heron.scheduler.TopologyRuntimeManagementException;
+import com.twitter.heron.spi.utils.NetworkUtils;
+
+/**
+ *  Utility functions for talking to a JSON-based scheduler API
+ */
+public final class SchedulerJsonAPIUtils {
```

Comment:
I recommend we put this in `c.t.h.common.network` since it's generic and not specific to schedulers. Also instead of a bunch of static helpers, could follow an approach more like the java network libraries or `HeronClient`, where we initialize a `HttpJsonClient` object with a base uri, and then call one of get, post, put delete on it? That way we can provide better control/contact over the connection lifecycle. It also makes it easier to use in unit tests via mocks.

We could throw `IOException` instead of `TopologyRuntimeManagementException`.